Clarify which point is encoded as x

emlun opened the pull request:
Prompted by @BasileiosKal's review comment in PR #31: https://github.com/tplooker/draft-ietf-cose-bls-key-representations/pull/31#discussion_r2487162512

I also took the opportunity for a couple related touches:
- Add `and in (#encoding-bls48-581)` to the two other places referencing `Appendix B.2 of [@BBS]`.
- Delete redundant "normatively" from "as normatively defined".

emlun committed:
Clarify which point is encoded as x

Prompted by review comment in PR #31:
https://github.com/tplooker/draft-ietf-cose-bls-key-representations/pull/31#discussion_r2487162512

I also took the opportunity for a couple related touches:
- Add "and in (#encoding-bls48-581)" to the two other places referencing
  Appendix B.2 of [@BBS].
- Delete redundant "normatively" from "as normatively defined".

selfissued commented on draft-ietf-cose-bls-key-representations.md:
```diff
@@ -75,7 +75,7 @@ When expressing a cryptographic key for these curves in JSON Web Key (JWK) form,
 - The parameter "crv" MUST be present and value MUST be one defined in (#curve-parameter-registration).
-- The parameter "x" MUST be present with its value being the base64url encoding of the compressed serialized point format defined normatively in Appendix B of [@BBS].
+- The parameter "x" MUST be present with its value being the base64url encoding of the compressed public key curve point, serialized as defined in Appendix B.2 of [@BBS] and in (#encoding-bls48-581).
 - The parameter "d" MUST be present for private key representations whose value MUST contain the big-endian representation of the private key base64url encoded without padding as defined in [@!RFC7515] Appendix C. This parameter MUST NOT be present for public keys.
```
```suggestion
- The parameter "d" MUST be present for private key representations whose value MUST contain the big-endian representation of the private key base64url encoded without padding as defined in [@!RFC7515] Appendix C. The length of this octet string MUST be `ceiling(log-base-2(n)/8)` octets (where `n` is the order of the curve). This parameter MUST NOT be present for public keys.
```

The length requirement was somehow lost.

BasileiosKal replied:
Just a note that `n is the order of the curve` is not completely accurate. In those curves, secret keys are moded with the order `r` of the larger subgroup of the curve (which is a prime divisor of the curve order).

I would be happy to open a PR with the clarification, if it would help.

emlun replied:
Looks like it was lost in commit cacdfb341f17ede04177e64a09919c840a3597cb that merged from #31 into #32: see diff https://github.com/tplooker/draft-ietf-cose-bls-key-representations/compare/4d37d8ea56fcf7ffc71a0017077370ec6c8ebfe0...cacdfb341f17ede04177e64a09919c840a3597cb

@BasileiosKal Oh right, that's an excellent correction. Will fix! It should be enough to just change "curve" to "curve generator", right?

emlun replied:
Fixed in 1f7edcccbe9e5dcbd1dedb76df77d89f80c8e52c and 089e090eafb81c2dd4a3f842e8054fce23f7bfa5.

selfissued commented on draft-ietf-cose-bls-key-representations.md:
```diff
@@ -84,7 +84,7 @@ When expressing a cryptographic key for these curves in COSE_Key form, the follo
 - The parameter "crv" (-1) MUST be present and value MUST be one defined in (#curve-parameter-registration).
-- The parameter "x" (-2) MUST be present with its value being the compressed serialized point format defined normatively in Appendix B of [@BBS].
+- The parameter "x" (-2) MUST be present with its value being the compressed public key curve point, serialized as defined in Appendix B.2 of [@BBS] and in (#encoding-bls48-581).
 - The parameter "d" (-4) MUST be present for private key representations whose value MUST contain the big-endian representation of the private key. This parameter MUST NOT be present for public keys.
```
```suggestion
- The parameter "d" (-4) MUST be present for private key representations whose value MUST contain the big-endian representation of the private key. The length of this octet string MUST be `ceiling(log-base-2(n)/8)` octets (where `n` is the order of the curve).  This parameter MUST NOT be present for public keys.
```

The length text was somehow lost.

emlun replied:
Fixed in 1f7edcccbe9e5dcbd1dedb76df77d89f80c8e52c.

emlun committed:
Restore fixed-length requirement lost in commit cacdfb3

This text was lost in the merge commit, see:

```
$ git diff cacdfb341f17ede04177e64a09919c840a3597cb^2 cacdfb341f17ede04177e64a09919c840a3597cb
```

emlun committed:
Fix incorrect reference to curve order

The length of `d` is defined by the order of the largest prime order subgroup,
not necessarily the order of the curve (these coincide when the curve is of
prime order, but these BLS curves are not of prime order). Or, equivalently, the
order of the generator point chosen for that subgroup.

Also change `n` to `r` to align with the notation in pairing-friendly-curves:
https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-12#name-selection-of-pairing-friend

emlun committed:
Add -09 change history

emlun committed:
Add Vasilis Kalos to Acknowledgements

emlun commented:
@selfissued I'll fix the merge conflict, give me a minute!

emlun committed:
Merge branch 'main' into clarify-x-point
